File: Arrays/contains-duplicate-ii.cs
// 219. Contains Duplicate II
// https://leetcode.com/problems/contains-duplicate-ii/description/

public class Solution {
    public bool ContainsNearbyDuplicate(int[] nums, int k) {
        var dict = new Dictionary<int, int>();
        for(int i=0; i  < nums.Length; i++){
            if(dict.ContainsKey(nums[i]))
                if(i - dict[nums[i]] <= k) return true;
            dict[nums[i]] = i;
        }
        return false;
    }
}

File: Arrays/contains-duplicate-ii.py
# 219. Contains Duplicate II
# https://leetcode.com/problems/contains-duplicate-ii/description/

class Solution(object):
    def containsNearbyDuplicate(self, nums, k):
        dict = {}
        for i, item in enumerate(nums):
            if item in dict:
                if i - dict[item] <= k: return True
            dict[item] = i
        return False